File: app/src/main/java/com/qxkqyy/activity/MainActivity.java
package com.qxkqyy.activity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Toast;

import com.donkingliang.banner.CustomBanner;
import com.facebook.drawee.backends.pipeline.Fresco;
import com.facebook.drawee.drawable.ScalingUtils;
import com.facebook.drawee.generic.GenericDraweeHierarchy;
import com.facebook.drawee.view.SimpleDraweeView;
import com.qxkqyy.R;
import com.qxkqyy.adapter.AppointmentAdapter;
import com.qxkqyy.bean.AppointmentModel;
import com.qxkqyy.utils.RealmHelper;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

import kotlin.TypeCastException;
import kotlin.jvm.internal.Intrinsics;

/**
 * Created by puyihao on 2018/1/1.
 */

public class MainActivity extends AppCompatActivity {
    private CustomBanner mBanner;
    private AppointmentAdapter mAdapter;
    private RecyclerView mRvAppointment;
    private List<AppointmentModel> mAppointmentModelList;
    private int ADD_APPOINTMENT = 100;

    protected void onCreate(@Nullable Bundle savedInstanceState) {
        Fresco.initialize((Context)this);
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_main);
        initWidget();
        initData();
    }

    protected void onStart() {
        super.onStart();
        initBanner();
    }

    private void initWidget(){
        View account = this.findViewById(R.id.account);
        account.setOnClickListener((View.OnClickListener)(new View.OnClickListener() {
            public final void onClick(View it) {
                startActivity(new Intent(getApplicationContext(), Login2Activity.class));
                overridePendingTransition(R.anim.leftout, R.anim.leftin);
            }
        }));
        View account2 = this.findViewById(R.id.account2);
        account2.setOnClickListener((View.OnClickListener)(new View.OnClickListener() {
            public final void onClick(View it) {
                startActivity(new Intent(getApplicationContext(), AccountActivity.class));
                overridePendingTransition(R.anim.leftout, R.anim.leftin);
            }
        }));
        View drlist = this.findViewById(R.id.doctorList);
        drlist.setOnClickListener((View.OnClickListener)(new View.OnClickListener() {
            public final void onClick(View it) {
                startActivity(new Intent(getApplicationContext(), ListActivity.class));
                overridePendingTransition(R.anim.upout, R.anim.upin);
            }
        }));
        View chat = this.findViewById(R.id.chat);
        chat.setOnClickListener((View.OnClickListener)(new View.OnClickListener() {
            public final void onClick(View it) {
                SharedPreferences preferences = getSharedPreferences("user", 0);
                String logged = preferences.getString("logged", "0");
                if(Intrinsics.areEqual(logged, "1")) {
                    startActivity(new Intent(getApplicationContext(), BoardActivity.class));
                    overridePendingTransition(R.anim.upout, R.anim.upin);
                } else {
                    Toast.makeText(getApplicationContext(), (CharSequence)"请登录后再留言！", Toast.LENGTH_SHORT).show();
                }

            }
        }));
        View newtask = this.findViewById(R.id.newTask);
        newtask.setOnClickListener((View.OnClickListener)(new View.OnClickListener() {
            public final void onClick(View it) {
                SharedPreferences preferences = getSharedPreferences("user", 0);
                String logged = preferences.getString("logged", "0");
                if(Intrinsics.areEqual(logged, "1")) {
                    startActivityForResult(new Intent(getApplicationContext(), NewActivity.class),ADD_APPOINTMENT);
                    overridePendingTransition(R.anim.upout, R.anim.upin);
                } else {
                    Toast.makeText(getApplicationContext(), (CharSequence)"请登录后再新建预约！", Toast.LENGTH_SHORT).show();
                }

            }
        }));

        mRvAppointment = (RecyclerView) findViewById(R.id.rv_content);
        mRvAppointment.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
        mRvAppointment.setItemAnimator(new DefaultItemAnimator());
    }

    private void initData(){
        mAppointmentModelList = RealmHelper.getRealm(getApplicationContext()).getAppointment();
        mAdapter = new AppointmentAdapter(getApplicationContext(),mAppointmentModelList);
        mRvAppointment.setAdapter(mAdapter);
    }

    public void initBanner(){
        View banner = this.findViewById(R.id.banner);
        if(banner == null) {
            throw new TypeCastException("null cannot be cast to non-null type com.donkingliang.banner.CustomBanner<kotlin.String>");
        } else {
            this.mBanner = (CustomBanner)banner;
            List images = (List)(new ArrayList());
            images.add("https://1970633640.github.io/webapp/banner1.png");
            images.add("https://1970633640.github.io/webapp/banner2.jpg");
            this.setBean(images);
            View account = this.findViewById(R.id.account);
            View account2 = this.findViewById(R.id.account2);
            View my1 = this.findViewById(R.id.my1);
            SharedPreferences preferences = this.getSharedPreferences("user", 0);
            String logged = preferences.getString("logged", "0");
            if(Intrinsics.areEqual(logged, "0")) {
                account.setVisibility(View.VISIBLE);
                account2.setVisibility(View.GONE);
                my1.setVisibility(View.GONE);
                mRvAppointment.setVisibility(View.GONE);
            }

            if(Intrinsics.areEqual(logged, "1")) {
                account2.setVisibility(View.VISIBLE);
                my1.setVisibility(View.VISIBLE);
                mRvAppointment.setVisibility(View.VISIBLE);
                account.setVisibility(View.GONE);
            }

            CustomBanner banner1 = this.mBanner;
            if(this.mBanner != null) {
                banner1.setOnPageClickListener((CustomBanner.OnPageClickListener)(new CustomBanner.OnPageClickListener() {
                    // $FF: synthetic method
                    // $FF: bridge method
                    public void onPageClick(int var1, Object var2) {
                        this.onPageClick(var1, (String)var2);
                    }

                    public final void onPageClick(int i, @NotNull String t) {
                        Intrinsics.checkParameterIsNotNull(t, "t");
                        startActivity(new Intent(getApplicationContext(), WebActivity.class));
                    }
                }));
            }
        }

    }

    private final void setBean(List beans) {
        CustomBanner customBanner = this.mBanner;
        if(this.mBanner != null) {
            customBanner = customBanner.setPages((CustomBanner.ViewCreator)(new CustomBanner.ViewCreator() {
                @NotNull
                public View createView(@NotNull Context context, int position) {
                    Intrinsics.checkParameterIsNotNull(context, "context");
                    SimpleDraweeView imageView = new SimpleDraweeView(context);
                    ((GenericDraweeHierarchy)imageView.getHierarchy()).setPlaceholderImage(R.drawable.picture, ScalingUtils.ScaleType.CENTER_INSIDE);
                    return (View)imageView;
                }

                public void updateUI(@NotNull Context context, @NotNull View view, int position, @NotNull String entity) {
                    Intrinsics.checkParameterIsNotNull(context, "context");
                    Intrinsics.checkParameterIsNotNull(view, "view");
                    Intrinsics.checkParameterIsNotNull(entity, "entity");
                    Uri uri1 = Uri.parse("android.resource://" + context.getPackageName() + "/" + R.drawable.picture);
                    Uri uri = Uri.parse(entity);
                    SimpleDraweeView draweeView = (SimpleDraweeView)view;
                    draweeView.setImageURI(uri);
                }

                // $FF: synthetic method
                // $FF: bridge method
                public void updateUI(Context var1, View var2, int var3, Object var4) {
                    this.updateUI(var1, var2, var3, (String)var4);
                }
            }), beans);
            if(customBanner != null) {
                customBanner.setScrollDuration(100);
            }
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        mAppointmentModelList.clear();
        mAppointmentModelList.addAll(RealmHelper.getRealm(getApplicationContext()).getAppointment());
        mAdapter.notifyDataSetChanged();
        mRvAppointment.scrollToPosition(0);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == ADD_APPOINTMENT){
            onResume();
        }
    }
}

File: app/src/main/java/com/qxkqyy/MainActivity.kt
package com.qxkqyy

import android.content.Context
import android.net.Uri
import android.os.Build
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import com.facebook.drawee.backends.pipeline.Fresco
import com.donkingliang.banner.CustomBanner
import com.facebook.drawee.view.SimpleDraweeView
import android.provider.CalendarContract
import android.provider.CalendarContract.Events
import android.content.Intent

import com.facebook.drawee.drawable.ScalingUtils
import android.content.SharedPreferences
import android.widget.Toast


class MainActivity : AppCompatActivity() {
    private var mBanner: CustomBanner<String>? = null
    private fun setBean(beans: List<String>) {
        mBanner?.setPages(object : CustomBanner.ViewCreator<String> {
            override fun createView(context: Context, position: Int): View {
                val imageView = SimpleDraweeView(context)
                imageView.hierarchy.setPlaceholderImage(R.drawable.picture, ScalingUtils.ScaleType.CENTER_INSIDE)
                return imageView
            }

            override fun updateUI(context: Context, view: View, position: Int, entity: String) {
                val uri1 = Uri.parse("android.resource://" + context.packageName + "/" + R.drawable.picture)
                val uri = Uri.parse(entity)
                val draweeView = view as SimpleDraweeView
                draweeView.setImageURI(uri);
            }

        }, beans)
                //                //设置指示器为普通指示器
                //                .setIndicatorStyle(CustomBanner.IndicatorStyle.ORDINARY)
                //                //设置两个点图片作为翻页指示器，不设置则没有指示器，可以根据自己需求自行配合自己的指示器,不需要圆点指示器可用不设
                //                .setIndicatorRes(R.drawable.shape_point_select, R.drawable.shape_point_unselect)
                //                //设置指示器的方向
                //                .setIndicatorGravity(CustomBanner.IndicatorGravity.CENTER)
                //                //设置指示器的指示点间隔
                //                .setIndicatorInterval(20)
                //设置自动翻页
                ?.scrollDuration = 100
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        Fresco.initialize(this);
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
        val textView1 = findViewById(R.id.textView1)
        textView1.setOnClickListener {
            //            val intent = Intent(Intent.ACTION_INSERT)
//                    .setData(Events.CONTENT_URI)
//                    .putExtra(Events.TITLE, "把预约添加到日程表")
//                    .putExtra(Events.EVENT_LOCATION, "预约的地点")
//                    .putExtra(CalendarContract.EXTRA_EVENT_BEGIN_TIME, 1513739532000)//unix时间戳毫秒
//                    .putExtra(CalendarContract.EXTRA_EVENT_END_TIME, 1513743132000)
//            if (intent.resolveActivity(packageManager) != null) {
//                startActivity(intent)
//            }
            startActivity(Intent(overridePendingTransition@ this, ScheduleActivity::class.java))
            overridePendingTransition(R.anim.upout, R.anim.upin)
        }
        val account = findViewById(R.id.account)
        account.setOnClickListener {
            startActivity(Intent(overridePendingTransition@ this, Login2Activity::class.java))
            overridePendingTransition(R.anim.leftout, R.anim.leftin)
        }
        val account2 = findViewById(R.id.account2)
        account2.setOnClickListener {
            startActivity(Intent(overridePendingTransition@ this, AccountActivity::class.java))
            overridePendingTransition(R.anim.leftout, R.anim.leftin)
        }
        val drlist = findViewById(R.id.doctorList)
        drlist.setOnClickListener {
            startActivity(Intent(overridePendingTransition@ this, ListActivity::class.java))
            overridePendingTransition(R.anim.upout, R.anim.upin)
        }
        val chat = findViewById(R.id.chat)
        chat.setOnClickListener {
            val preferences = getSharedPreferences("user", 0)
            val logged = preferences.getString("logged", "0")
            if (logged == "1") {
                startActivity(Intent(overridePendingTransition@ this, BoardActivity::class.java))
                overridePendingTransition(R.anim.upout, R.anim.upin)
            } else {
                Toast.makeText(getApplicationContext(), "请登录后再留言！",
                        Toast.LENGTH_SHORT).show();
            }
        }
        val newtask = findViewById(R.id.newTask)
        newtask.setOnClickListener {
            val preferences = getSharedPreferences("user", 0)
            val logged = preferences.getString("logged", "0")
            if (logged == "1") {
                startActivity(Intent(overridePendingTransition@ this, NewActivity::class.java))
                overridePendingTransition(R.anim.upout, R.anim.upin)
            } else {
                Toast.makeText(getApplicationContext(), "请登录后再新建预约！",
                        Toast.LENGTH_SHORT).show();
            }
        }
    }

    override fun onStart() {
        super.onStart()
        mBanner = findViewById(R.id.banner) as CustomBanner<String>
        val images = mutableListOf<String>()
        images.add("https://1970633640.github.io/webapp/banner1.png")
        images.add("https://1970633640.github.io/webapp/banner2.jpg")
        setBean(images)
        val account = findViewById(R.id.account)
        val account2 = findViewById(R.id.account2)
        val my1 = findViewById(R.id.my1)
        val my2 = findViewById(R.id.textView1)
        val preferences = getSharedPreferences("user", 0)
        val logged = preferences.getString("logged", "0")
        if (logged == "0") {
            account.visibility = View.VISIBLE
            account2.visibility = View.GONE
            my1.visibility = View.GONE;my2.visibility = View.GONE
        }
        if (logged == "1") {
            account2.visibility = View.VISIBLE
            my1.visibility = View.VISIBLE
            my2.visibility = View.VISIBLE
            account.visibility = View.GONE
        }
        mBanner?.setOnPageClickListener(CustomBanner.OnPageClickListener
        { i: Int, t: String -> startActivity(Intent(this, WebActivity::class.java)) })
    }

}
